--- ming-0.4.4/src/blocks/gifdbl.c.bero	2013-03-01 12:52:28.083494178 +0100
+++ ming-0.4.4/src/blocks/gifdbl.c	2013-03-01 12:52:46.148268337 +0100
@@ -227,7 +227,7 @@ SWFDBLBitmapData newSWFDBLBitmapData_fro
 	SWFDBLBitmapData ret;
 	struct dbl_data gifdata;
 
-	if((file = DGifOpenFileName(fileName)) == NULL)
+	if((file = DGifOpenFileName(fileName, NULL)) == NULL)
 		return NULL;
 	if(!readGif(file, &gifdata))
 		return NULL;
@@ -246,7 +246,7 @@ SWFDBLBitmapData newSWFDBLBitmapData_fro
 	SWFDBLBitmapData ret;
 	struct dbl_data gifdata;
 
-	if((file = DGifOpen(input, (InputFunc) gifReadFunc)) == NULL)
+	if((file = DGifOpen(input, (InputFunc) gifReadFunc, NULL)) == NULL)
 		return NULL;
 	if(!readGif(file, &gifdata))
 		return NULL;
--- ming-0.4.4/util/gif2dbl.c.bero	2013-03-01 13:16:41.037329755 +0100
+++ ming-0.4.4/util/gif2dbl.c	2013-03-01 13:17:21.940818391 +0100
@@ -17,10 +17,12 @@
 #include <gif_lib.h>
 #include <zlib.h>
 
+static int gif_error = 0;
+
 void error(char *msg)
 {
   printf("%s:\n\n", msg);
-  PrintGifError();
+  printf("%s\n", GifErrorString(gif_error));
   exit(-1);
 }
 
@@ -57,7 +59,7 @@ unsigned char *readGif(char *fileName, i
   unsigned char *p;
   int i, nColors, size, alpha, bgColor, alignedWidth;
 
-  if((file = DGifOpenFileName(fileName)) == NULL)
+  if((file = DGifOpenFileName(fileName, &gif_error)) == NULL)
     error("Error opening file");
 
   if(DGifSlurp(file) != GIF_OK)
--- ming-0.4.4/util/gif2mask.c.bero	2013-03-01 13:15:07.236502427 +0100
+++ ming-0.4.4/util/gif2mask.c	2013-03-01 13:16:32.868431881 +0100
@@ -11,10 +11,12 @@
 
 #define max(a,b,c) (((a)>(b))?(((c)>(a))?(c):(a)):(((c)>(b))?(c):(b)))
 
+static int gif_error = 0;
+
 void error(char *msg)
 {
   printf("%s:\n\n", msg);
-  PrintGifError();
+  printf("%s\n", GifErrorString(gif_error));
   exit(-1);
 }
 
@@ -26,7 +28,7 @@ unsigned char *readGif(char *fileName, i
   unsigned char *data;
   int i, nColors, size;
 
-  if((file = DGifOpenFileName(fileName)) == NULL)
+  if((file = DGifOpenFileName(fileName, &gif_error)) == NULL)
     error("Error opening file");
 
   if(DGifSlurp(file) != GIF_OK)
